src/CalcViewModel/Common/Utils.h

```diff
@@ -32,6 +32,15 @@
     }}\
     } private: t m_##n; public:
 
+#define OBSERVABLE_PROPERTY_RW_ALWAYS_NOTIFY(t, n)\
+    property t n {\
+    t get() { return m_##n; }\
+    void set(t value) {\
+    m_##n = value;\
+    RaisePropertyChanged(L#n);\
```

Comment:
What are we trying to fix with it this macro? Why is it important to launch `PropertyChanged` when the value didn't change?

Reply:
I added this in because it handles a lot of work that I would need to do manually otherwise when binding a TextBox Text property to this property. For example, if a user types invalid input, this will let me set the Value to the previous Value and still fire the event, causing the binding to update, resetting the double in the textbox to the previous value with correct formatting.

I could just manually set the TextBox text in SubmitTextbox instead, but this would require extra work to also format the double -> string conversion to remove trailing zeroes. I noticed you added a utility in one of your recent PRs that lets us remove trailing zeroes very easily from a string. I added a comment to go ahead and remove this work around and use your util function once we re-merge with master and pick up that change.